scipy/stats/tests/test_hypotests.py

```diff
@@ -994,6 +995,29 @@ def test_somersd_perfect_correlation(self, positive_correlation):
         assert res.statistic == expected_statistic
         assert res.pvalue == (0 if positive_correlation else 1)
 
+    def test_somersd_large_inputs(self):
+        # Test that large inputs where potential overflows could occur give
+        # the expected output. This is tested in the case of binary inputs.
+
+        # generate lists of random classes 1-2 (binary)
+        classes = [1, 2]
+        n_samples = 10 ** 6
+        random.seed(6272161)
+        x = random.choices(classes, k=n_samples)
+        y = random.choices(classes, k=n_samples)
+
+        # get value to compare with: sklearn output
+        # from sklearn import metrics
+        # val_auc_sklearn = metrics.roc_auc_score(x, y)
+        # # convert to the Gini coefficient (Gini = (AUC*2)-1)
+        # val_sklearn = 2 * val_auc_sklearn - 1
+        val_sklearn = -0.001528138777036947
+
+        # calculate the Somers' D statistic, which should be equal to the result
+        # of val_sklearn until approximately machine precision
```

Comment:
```suggestion
        # calculate the Somers' D statistic, which should be equal to the
        # result of val_sklearn until approximately machine precision
```
Line too long (PEP8)